flank-scripts/src/main/kotlin/flank/scripts/testartifacts/core/DownloadFixtures.kt

```diff
@@ -0,0 +1,53 @@
+package flank.scripts.testartifacts.core
+
+import com.jcabi.github.Release
+import flank.scripts.github.getRelease
+import flank.scripts.utils.download
+import okhttp3.OkHttpClient
+import java.io.File
+
+fun Context.downloadFixtures(
+    overwrite: Boolean = false
+) {
+    downloadFixtures(
+        url = latestArtifactZipUrl(branch),
+        testArtifacts = projectRoot.testArtifacts,
+        overwrite = overwrite
+    )
+}
+
+private fun Context.downloadFixtures(
+    testArtifacts: File,
+    url: String,
+    overwrite: Boolean
+) {
+    print("* Downloading test artifacts from url $url - ")
+    testArtifacts.resolve(
+        ArtifactsArchive(
+            branch = branch,
+            timestamp = url.parseArtifactsArchive().timestamp
+        ).fullName
+    ).run {
+        if (exists() && overwrite) delete()
+        if (!exists()) OkHttpClient().download(url, absolutePath).also { println("OK") }
+        else println("ABORTED (already exists)")
+    }
+}
+
+private fun latestArtifactZipUrl(branch: String): String =
+    requireNotNull(
+        testArtifactsRepo().getRelease(branch)
+    ) { "Cannot find release for branch $branch" }.run {
+        requireNotNull(
+            latestArtifactZipUrl()
+        ) { "Cannot find artifacts download url for release ${name()}" }
+    }
```

Comment:
I really do not like this formatting

Reply:
That's my way, suggest yours, then we will compare.

Reply:
how about 
```kotlin
    requireNotNull(testArtifactsRepo().getRelease(branch)) { "Cannot find release for branch $branch" }
        .run {
            requireNotNull(latestArtifactZipUrl()) { "Cannot find artifacts download url for release ${name()}" }
        }
```

or even
```kotlin
    requireNotNull(testArtifactsRepo().getRelease(branch)) { "Cannot find release for branch $branch" }
        .run { requireNotNull(latestArtifactZipUrl()) { "Cannot find artifacts download url for release ${name()}" } }
```


If you prefer yours, then just leave it :)

Reply:
I prefere to leave it ;).